### DISCUSSION TITLE
ScriptReference/ParticleSystem.MinMaxCurve

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#403] [Notes]
# ScriptReference/ParticleSystem.MinMaxCurve

https://docs.unity3d.com/ScriptReference/ParticleSystem.MinMaxCurve.html

**Comment by Carwashh** [comment]
I've been trying to create a Timeline playable to manipulate this, the particle system that's setup required 'Random between two Constants', once getting the StartLifeTime updating via the timeline working I found that the ConstantMax value was always double of what I wanted it to be. An hour or so later I found out constant and constantMax are not treated separately, so doing the following was setting the value twice.  

```csharp
curveA.constant += curveB.constant * weight;  
curveA.constantMin += curveB.constantMin * weight;  
curveA.constantMax += curveB.constantMax * weight;
```
